front_end/panels/rn_welcome/RNWelcome.ts

```diff
@@ -96,9 +98,16 @@ export class RNWelcomeImpl extends UI.Widget.VBox implements SDK.TargetManager.S
   }
 
   private _updateReactNativeVersion(event: Common.EventTarget.EventTargetEvent<Protocol.ReactNativeApplication.MetadataUpdatedEvent>): void {
+    const {appDisplayName, deviceName, reactNativeVersion} = event.data;
+
+    if (reactNativeVersion != null) {
+      this.reactNativeVersion = reactNativeVersion === '1000.0.0' ? 'main' : reactNativeVersion;
```

Comment:
I believe this logic should live in React Native, not in Fusebox frontend, what do you think?

Reply:
It'll take some work, depending on the level it's implemented at, as many scripts within the RN repo and fbsource depend on `1000.0.0`. I'd also like our CDP server to unambiguously report the version that the codebase has (`ReactNativeVersion.java` etc) — I consider this a UI flourish!

Reply:
No, but can you actually do this logic right before sending the information to Fusebox frontend? Where you propagating and setting information about the host.